cfgov/core/tests/test_middleware.py

```diff
@@ -0,0 +1,29 @@
+import unittest
+
+from django.test import TestCase, override_settings
+
+import mock
+
+from v1.models.browse_filterable_page import BrowseFilterablePage
+from v1.tests.wagtail_pages.helpers import publish_page
+
+
+class TestParseLinksMiddleware(TestCase):
+
+    @mock.patch('core.utils.parse_links')
+    def test_parse_links_gets_called(self, mock_parse_links):
+        """Middleware correctly invokes parse links when rendering webpage"""
+        bfp = BrowseFilterablePage(
+            title='Browse Filterable Page',
+            slug='random-page',
+        )
+        publish_page(child=bfp)
+        response = self.client.get('/random-page')
+        mock_parse_links.assert_called_with(response.content)
+
+    @override_settings(PARSE_LINKS_BLACKLIST=['/foo/'])
+    @mock.patch('core.utils.parse_links')
+    def test_parse_links_does_not_get_called(self, mock_parse_links):
+        """Middleware does not invoke parse links when path is in blacklist"""
+        self.client.get('/foo/bar')
```

Comment:
@chosak I started writing tests for this and am a little stuck.  If you run this as is, the first test fails with 
```
AssertionError: Expected call: parse_links('')
Not called
```
I couldn't quite figure out why because as I was stepping through it, it was indeed getting called.  Then I removed the second test altogether and the first test passed.  So then I went through line by line and noticed that if line 28 (in the second test) is commented out, the first test passes. This seems odd because in the docs, https://docs.djangoproject.com/en/2.0/topics/testing/tools/#django.test.SimpleTestCase.client, it means that the client gets recreated in each test.  Any thoughts??

Reply:
@richaagarwal if you change the patch to `@mock.patch('core.middleware.parse_links')`, it works. I think you have to patch the module path as encountered by the test, not necessarily where it is imported from in the end. See "Patching in the wrong place" [here](http://alexmarandon.com/articles/python_mock_gotchas/).

Reply:
@chosak ah, thanks so much!  I think I've probably encountered that before but never realized that was what was going on.